agent/lib/kontena/workers/image_pull_worker.rb

```diff
@@ -4,6 +4,8 @@ class ImagePullWorker
     include Celluloid
     include Kontena::Logging
 
+    exclusive :ensure_image
```

Comment:
Oh, right. I agree that your example is simpler. But we can probably refactor this later.